dipy/reconst/bingham.py

```diff
@@ -0,0 +1,256 @@
+"""Tools for fitting Bingham distributions to orientation distributions,
+as described in Riffert et al [1]_. The resulting distributions can further
+be used to compute ODF-lobe-specific measures such as the fiber density
+and fiber spread [1]_.
+
+References
+----------
+.. [1] Riffert TW, Schreiber J, Anwander A, Knösche TR. Beyond fractional
+           anisotropy: Extraction of bundle-specific structural metrics from
+           crossing fiber models. NeuroImage. 2014 Oct 15;100:176-91.
+"""
+
+from math import cos, radians
+from warnings import warn
+import numpy as np
+
+from dipy.direction import peak_directions
+
+
+def bingham_to_sf(f0, k1, k2 , major_axis, minor_axis, vertices):
+    """
+    Evaluate Bingham distribution for a sphere.
+
+    Parameters
+    ----------
+    f0: float
+        Maximum amplitude of the distribution.
+    k1: float
+        Concentration along major axis.
+    k2: float
+        Concentration along minor axis.
+    vertices: ndarray (N, 3)
+        Unit sphere directions along which the distribution
+        is evaluated.
+    """
+    if not (np.linalg.norm(vertices, axis=-1) == 1).any():
+        warn("Some sphere directions are not normalized. Normalizing.",
+             UserWarning)
+        vertices /= np.linalg.norm(vertices, axis=-1, keepdims=True)
+
+    sf = f0*np.exp(-k1*vertices.dot(major_axis)**2
+                   -k2*vertices.dot(minor_axis)**2)
+    return sf.T
+
+
+def bingham_fit_sf(sf, sphere, max_search_angle=15,
```

Comment:
The same for this name, can you remove the 'sf' from this function name? this is for clarity so that it's not to be confused with the FS output parameter.